### PR TITLE
Add test for scientific python packages

### PR DESCRIPTION
**Pull request by grisu48**
This commit adds a test for numpy and scipy packages on openSUSE and
SLE>=15

- Related ticket: https://progress.opensuse.org/issues/81376
- Verification runs: [Tumbleweed](http://duck-norris.qam.suse.de/tests/4457) | [SLE15-SP2](http://duck-norris.qam.suse.de/tests/4460) | [SLE15-SP1](http://duck-norris.qam.suse.de/tests/4459) | [SLE15](http://duck-norris.qam.suse.de/tests/4458)

### CODE REVIEW

File: tests/console/python_scientific.pm [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        add_suseconnect_product(get_addon_fullname('phub'));
+    }
+    zypper_call 'in python3 python3-numpy python3-scipy';
+    # Run python scripts
+    run_python_script('python3-numpy-test.py');
+    run_python_script('python3-scipy-test.py');
+}
+
+sub cleanup() {
+    # unregister SDK
```
Comment (dzedro) [inline tests/console/python_scientific.pm:51]: is not sdk and comment not really needed

File: tests/console/python_scientific.pm [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    if (script_run("grep 'Softfail' $logfile") == 0) {
+        my $failmsg = script_output("grep 'Softfail' '$logfile'");
+        record_soft_failure("$failmsg");
+    }
+}
+
+sub run {
+    my $self = shift;
+    select_console 'root-console';
+    #$self->select_serial_terminal;
```
Comment (dzedro) [inline tests/console/python_scientific.pm:38]: Why not serial console ?

Comment (grisu48) [reply tests/console/python_scientific.pm]: Updated

File: tests/console/python_scientific.pm [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+# * Test scipy
+# Maintainer: Felix Niederwanger <felix.niederwanger@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils;
+use registration qw(cleanup_registration register_product add_suseconnect_product get_addon_fullname remove_suseconnect_product);
```
Comment (dzedro) [inline tests/console/python_scientific.pm:21]: From `version_utils` you use only `is_sle`, people tend to don't define the list of subroutines if it's too long.
Do as you like. :)

File: lib/main_common.pm [hunk 4]
```diff
@@ -1692,9 +1692,10 @@ sub load_extra_tests_console {
     loadtest 'console/osinfo_db' if (is_sle('12-SP3+') && !is_jeos);
     loadtest 'console/libgcrypt' if ((is_sle(">=12-SP4") && (check_var_array('ADDONS', 'sdk') || check_var_array('SCC_ADDONS', 'sdk'))) || is_opensuse);
     loadtest "console/gd";
-    loadtest 'console/valgrind'       unless is_sle('<=12-SP3');
-    loadtest 'console/sssd_samba'     unless (is_sle("<15") || is_sle(">=15-sp2") || is_leap('>=15.2') || is_tumbleweed);
-    loadtest 'console/wpa_supplicant' unless (!is_x86_64 || is_sle('<15') || is_leap('<15.1') || is_jeos || is_public_cloud);
+    loadtest 'console/valgrind'          unless is_sle('<=12-SP3');
+    loadtest 'console/sssd_samba'        unless (is_sle("<15") || is_sle(">=15-sp2") || is_leap('>=15.2') || is_tumbleweed);
+    loadtest 'console/wpa_supplicant'    unless (!is_x86_64 || is_sle('<15') || is_leap('<15.1') || is_jeos || is_public_cloud);
+    loadtest 'console/python-scientific' unless (is_sle("<15"));
```
Comment (dzedro) [inline lib/main_common.pm:1698]: Just curious when this, I guess for somebody unnecessary complicated, unless condition gang started :)

**Review by dzedro — APPROVED** [review]

File: tests/console/python_scientific.pm [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+sub cleanup() {
+    if (is_sle) {
+        remove_suseconnect_product(get_addon_fullname('phub'));
+    }
+}
+
+sub post_fail_hook {
+    cleanup();
```
Comment (pdostal) [inline tests/console/python_scientific.pm:56]:
```suggestion
    cleanup();
    $self->SUPER::post_fail_hook;
```

File: tests/console/python_scientific.pm [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils qw(is_sle);
+use registration qw(cleanup_registration register_product add_suseconnect_product get_addon_fullname remove_suseconnect_product);
+
+sub run_python_script {
+    my $script  = shift;
+    my $logfile = "output.txt";
```
Comment (pdostal) [inline tests/console/python_scientific.pm:25]:
```suggestion
    my $logfile = "output.txt";
    record_info($script, "Running $script python script");
```

File: tests/console/python_scientific.pm [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        remove_suseconnect_product(get_addon_fullname('phub'));
+    }
+}
+
+sub post_fail_hook {
+    cleanup();
+}
+
+sub post_run_hook {
+    cleanup();
```
Comment (pdostal) [inline tests/console/python_scientific.pm:60]:
```suggestion
    cleanup();
    $self->SUPER::post_run_hook;
```

**Review by pdostal — CHANGES_REQUESTED** [review]

File: tests/console/python_scientific.pm (hunk 6 above)
Comment (grisu48) [reply tests/console/python_scientific.pm:25]: This is a good idea!

File: tests/console/python_scientific.pm (hunk 7 above)
Comment (pdostal) [reply tests/console/python_scientific.pm:63]: WDYT?

File: tests/console/python_scientific.pm (hunk 5 above)
Comment (grisu48) [reply tests/console/python_scientific.pm:58]: Done

**Review by pdostal — APPROVED** [review]

**Comment by Vogtinator** [comment]
This broke main_common.pm: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11712

**Comment by ostruzina** [comment]
It looks like it needs a little bit more looking into: https://openqa.suse.de/tests/5259741#step/python_scientific/19

(results from SLE15-SP1 and SLE15-SP2 are not available yet because the test did not run due to a broken aggregate build)

**Pull request merged**